### PR TITLE
*: readme updates for v2

### PR DESCRIPTION
**Pull request by ericchiang**
Final couple of changes before we switch dev to master.

cc @ElijahCaine @rithujohn191

### CODE REVIEW

File: Documentation/getting-started.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+## Configuration
+
+Dex is a single, scalable binary that pulls all configuration from a config file (no command line flags). Use the example config file defined in the `examples` folder to start up dex with an in-memory data store.
+
+```
+./bin/dex serve examples/config-dev.yaml
+```
+
+The [example config](../examples/config-dev.yaml) file documents many of the configuration options through inline comments. For extra config options, look at that file.
```
Comment (rithujohn191) [inline Documentation/getting-started.md]: This link point to https://github.com/coreos/dex/blob/readme-docs-v2/examples/config-dev.yaml. Is it meant to be https://github.com/coreos/dex/blob/dev/examples/config-dev.yaml?

File: Documentation/getting-started.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+## Running a client
+
+To interact with dex, like any other OAuth2 provider, users first visit a client app and are then redirected to dex to login. Dex ships with a example client app (also built with the `make` command), for testing and demos.
+
+UThe example app's flags default to user the statically configured OAuth2 client credentials in the `config-dev.yaml` file. Run the example app will cause it to query dex's [discovery endpoint][oidc-discovery] and determine the OAuth2 endpoints.
```
Comment (rithujohn191) [inline Documentation/getting-started.md]: nit: typo UThe

Comment (ericchiang) [inline Documentation/getting-started.md]: thanks! updated

File: Documentation/getting-started.md (hunk 1 above)
Comment (ElijahCaine) [inline Documentation/getting-started.md]: This link should be a reference, not inline.

File: Documentation/getting-started.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# Getting started
+
+## Building the dex binary
+
+Dex requires a Go installation and a GOPATH configured. For setting up a Go workspace, refer to the [official documentation][go-setup]. Clone it down the correct place, and simply type `make` to compile the dex binary.
+
+```
+git clone https://github.com/coreos/dex.git $GOPATH/src/github.com/coreos/dex
```
Comment (ElijahCaine) [inline Documentation/getting-started.md]: Code-block commands should start with `$ `

File: Documentation/getting-started.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+cd $GOPATH/src/github.com/coreos/dex
+make
+```
+
+A containerized version of this document using [rkt][rkt-repo] or Docker will be added later.
```
Comment (ElijahCaine) [inline Documentation/getting-started.md]: This is confusing. Can you elaborate as to what that means or remove/comment it out?

File: Documentation/getting-started.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+## Configuration
+
+Dex is a single, scalable binary that pulls all configuration from a config file (no command line flags). Use the example config file defined in the `examples` folder to start up dex with an in-memory data store.
```
Comment (ElijahCaine) [inline Documentation/getting-started.md]: `examples` should link to the config being referenced to.

Comment (ElijahCaine) [inline Documentation/getting-started.md]: I would reword this to be:

----

Dex exclusively pulls configuration options from a config file.  Use the `[example config][example-config]` file found in the `examples/` directory to start an instance of dex with an in-memory data store.

File: Documentation/getting-started.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+```
+./bin/dex serve examples/config-dev.yaml
+```
+
+The [example config](../examples/config-dev.yaml) file documents many of the configuration options through inline comments. For extra config options, look at that file.
+
+## Running a client
+
+To interact with dex, like any other OAuth2 provider, users first visit a client app and are then redirected to dex to login. Dex ships with a example client app (also built with the `make` command), for testing and demos.
```
Comment (ElijahCaine) [inline Documentation/getting-started.md]: I would reword this to be:

----

Dex operates like most other OAuth2 providers.  Users are redirected from a client app to dex to login.  Dex ships with [...]

File: Documentation/getting-started.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+The example app's flags default to user the statically configured OAuth2 client credentials in the `config-dev.yaml` file. Run the example app will cause it to query dex's [discovery endpoint][oidc-discovery] and determine the OAuth2 endpoints.
```
Comment (ElijahCaine) [inline Documentation/getting-started.md]: I would reword this to be:

----

The example application pulls configuration options from `[config-dev.yml][config-dev]`. Running the example app [...]

File: Documentation/getting-started.md (hunk 4 above)
Comment (ericchiang) [inline Documentation/getting-started.md]: removed

File: Documentation/getting-started.md (hunk 7 above)
Comment (ericchiang) [inline Documentation/getting-started.md]: Going to add this statement a little earlier.

> Use the [example config][example-config] file found in the `examples/` directory to start an instance of dex with an in-memory data store and a set of predefined OAuth2 clients.

Then change this to

> By default, the example client is configured with the same OAuth2 credentials defined in `examples/config-dev.yaml` to talk to dex. Running the example app will cause it to query dex's [discovery endpoint][oidc-discovery] and determine the OAuth2 endpoints.

That work?
